camlibs/ptp2/config.c

```diff
@@ -9372,15 +9411,63 @@ _put_Sony_FocusMagnifySetting(CONFIG_PUT_ARGS)
 	if (y > SONY_TOUCH_MAX_Y) {
 		y = SONY_TOUCH_MAX_Y;
 	}
-#undef SONY_TOUCH_MAX_X
-#undef SONY_TOUCH_MAX_Y
+
 	xpropval.u64 = ((uint64_t)(magInteger*10 + magDecimal) << 32) + (x << 16) + y;
 
 	C_PTP (ptp_sony_setdevicecontrolvaluea (params, dpd->DevicePropCode, &xpropval, PTP_DTC_UINT64));
 	*alreadyset = 1;
 	return GP_OK;
 }
 
+static int
+_get_Sony_SpotFocusArea(CONFIG_GET_ARGS) {
+	gp_widget_new (GP_WIDGET_TEXT, _(menu->label), widget);
+	gp_widget_set_name (*widget, menu->name);
+
+	// Current value not available via Sony API
+	uint32_t value = 0;
+	gp_widget_set_value  (*widget, &value);
+
+	return GP_OK;
+}
+
+static int
+_put_Sony_SpotFocusArea(CONFIG_PUT_ARGS) {
+	PTPParams* params = &(camera->pl->params);
+	PTPPropValue xpropval;
+	char* xval;
+	uint32_t x=0,y=0;
+	uint64_t currentVal;
+
+	// Camera must be set to an autofocus mode and focusarea has to be set to one of the Spot Focus modes
+
+	currentVal = dpd->CurrentValue.u32;
+	x = (int)(currentVal >> 16 & 0xffff);
```

Comment:
please add additional brackets ((currentVal >> 16) & 0xfff)

Reply:
Ok, added.  Sorry, not sure why I was so stingy with the brackets! :)